src/services/persistent-data/persistent-data.service.ts

```diff
@@ -0,0 +1,230 @@
+import { Injectable } from '@angular/core';
```

Comment:
Changes are good. There are a few console.logs in there to remove. I have sent you an email as this would be a perfect use case for the Adapter design pattern here. Would just involve slight reorganizing the code (see my email)